src/solidlsp/language_servers/al_language_server.py

```diff
@@ -0,0 +1,1045 @@
+    ) -> str:
+        """
+        Setup runtime dependencies for AL Language Server and return the command to start the server.
+
+        This method handles the complete setup process:
+        1. Determines the correct platform-specific executable path
+        2. Checks if the AL extension is already installed
+        3. Downloads the extension from VS Code marketplace if needed
+        4. Configures executable permissions on Unix systems
+        5. Returns the properly formatted command string
+
+        The AL Language Server executable is located in different paths based on the platform:
+        - Windows: extension/bin/win32/Microsoft.Dynamics.Nav.EditorServices.Host.exe
+        - Linux: extension/bin/linux/Microsoft.Dynamics.Nav.EditorServices.Host
+        - macOS: extension/bin/darwin/Microsoft.Dynamics.Nav.EditorServices.Host
+        """
+        # Directory where the AL extension will be installed
+        # This is typically ~/.serena/ls_resources/al-extension
+        al_extension_dir = os.path.join(cls.ls_resources_dir(solidlsp_settings), "al-extension")
+
+        # Determine platform-specific executable path within the extension
+        # AL provides different binaries for each platform
+        system = platform.system()
+        if system == "Windows":
+            executable_relative_path = os.path.join("extension", "bin", "win32", "Microsoft.Dynamics.Nav.EditorServices.Host.exe")
+        elif system == "Linux":
+            executable_relative_path = os.path.join("extension", "bin", "linux", "Microsoft.Dynamics.Nav.EditorServices.Host")
+        elif system == "Darwin":
+            executable_relative_path = os.path.join("extension", "bin", "darwin", "Microsoft.Dynamics.Nav.EditorServices.Host")
+        else:
+            raise RuntimeError(f"Unsupported platform: {system}")
+
+        executable_path = os.path.join(al_extension_dir, executable_relative_path)
+
+        # Check if the extension is already installed
+        if not os.path.exists(executable_path):
+            logger.log(f"AL Language Server not found at {executable_path}. Downloading from VS Code marketplace...", logging.INFO)
+
+            # AL extension version - using latest stable version
+            AL_VERSION = "latest"
+            url = f"https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-dynamics-smb/vsextensions/al/{AL_VERSION}/vspackage"
+
+            # Try to download the extension
+            if cls._download_al_extension(logger, url, al_extension_dir):
+                logger.log("AL Extension downloaded successfully.", logging.INFO)
+            else:
+                logger.log("Failed to download AL extension, falling back to manual search.", logging.WARNING)
+                # Fall back to the old method of finding the extension
+                return cls._get_language_server_command_fallback(logger)
+
+        # Verify executable exists after download
+        if not os.path.exists(executable_path):
```

Comment:
verification should happen after download, raising then. There shouldn't be two calls to the fallback